airflow/models/dag.py

```diff
@@ -2115,15 +2151,18 @@ def bulk_write_to_db(cls, dags: Collection["DAG"], session=None):
             .group_by(DagRun.dag_id)
             .all()
         )
+        filelocs = []
 
         for orm_dag in sorted(orm_dags, key=lambda d: d.dag_id):
             dag = dag_by_ids[orm_dag.dag_id]
             if dag.is_subdag:
+                filelocs.append(dag.fileloc)
                 orm_dag.is_subdag = True
                 orm_dag.fileloc = dag.parent_dag.fileloc  # type: ignore
                 orm_dag.root_dag_id = dag.parent_dag.dag_id  # type: ignore
                 orm_dag.owners = dag.parent_dag.owner  # type: ignore
             else:
+                filelocs.append(dag.fileloc)
```

Comment:
Could we move this outside the if/else as we have the same command in both cases

Reply:
👍🏻